api/dataprocessor.go

```diff
@@ -166,8 +167,108 @@ func aggEvery(numPoints, maxPoints uint32) uint32 {
 	return (numPoints + maxPoints - 1) / maxPoints
 }
 
-// error is the error of the first failing target request
 func (s *Server) getTargets(reqs []models.Req) ([]models.Series, error) {
+	// split reqs into local and remote.
+	localReqs := make([]models.Req, 0)
+	remoteReqs := make(map[*cluster.Node][]models.Req)
+	for _, req := range reqs {
+		if req.Node == nil || req.Node == cluster.ThisNode {
```

Comment:
also I think we can simplify this function: instead of first splitting up the requests in two categories, and then for each category, do very similar logic. we can just iterate over all requests, for each do the wg.Add, and launch the goroutine where we check the err and series results, etc. but just call getTargetsLocal vs getTargetsRemote based on whether it's a local or remote node